lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -67,6 +67,10 @@ static int getBitWidthOrSentinel(Type type) {
       .Case<FlipType>([](FlipType flipType) {
         return getBitWidthOrSentinel(flipType.getElementType());
       })
+      .Case<sv::InterfaceType, sv::ModportType>([](Type) {
+        // Treat interfaces and modports as single-bit.
+        return 1;
```

Comment:
This may be a bit of a hack, but it seems to work. We need some way to support these types. By saying they're a single bit, we won't try to print the bit width for them.

Is there a nicer way to add support for these types? We are just using their symbolic names, so I'm not really sure how a bit width even makes sense.

Reply:
> Is there a nicer way to add support for these types? We are just using their symbolic names, so I'm not really sure how a bit width even makes sense.

It doesn't. Assuming the EmitVerilog.cpp code hasn't changed much since I last, the type output stuff was very untyped, bit-width centric and heavily tied into FIRRTL making it difficult to add typed support. Yeah, this is a hack but I can't think of a better way unless you want to embark on a refactoring. I think said refactoring should happen at some point but IMO it should be after the FIRRTL support is dropped.

@lattner What do you think?

Reply:
I agree that this is a hack and I'd prefer it to go away.  The caller of `getBitWidthOrSentinel` should deal with modports instead of just passing them in.